### PR TITLE
Target netstandard2.0 for Orleans.Serialization.Abstractions

### PR DESCRIPTION
**Pull request by jsteinich**
Enables adding `Orleans.Serialization` attributes to projects targeting `netstandard2.1`.

Some notes:

- ~~Going to `netstandard2.0` results in XML comment errors due to the reference to `ValueTask`. Seems a bit silly to add a conditional package reference just for a comment. Perhaps ignoring the error would make sense.~~
- Could avoid the multi-targeting, but didn't want to remove a target.
- The distinction between standard and multi target naming feels a bit arbitrary at this point, but I tried to match historical usage.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/7965)

### CODE REVIEW

**Comment by jsteinich** [comment]
Ended up just disabling the documentation warning since 2.0 seems more appropriate (and is what I need for desired application).

File: src/Orleans.Serialization.Abstractions/Annotations.cs [hunk 1]
```diff
@@ -46,6 +46,7 @@ public GenerateMethodSerializersAttribute(Type proxyBase, bool isExtension = fal
         /// <value>true if this instance is extension; otherwise, false.</value>
         public bool IsExtension { get; }
 
+#pragma warning disable 1574 // ValueTask not directly in netstandard2.0
```
Comment (ReubenBond) [inline src/Orleans.Serialization.Abstractions/Annotations.cs:49]: Should we be adding a reference to the ValueTask package for netstandard2.0?

**Comment by ReubenBond** [comment]
>The distinction between standard and multi target naming feels a bit arbitrary at this point, but I tried to match historical usage.

Yes, definitely. It made sense originally, but not anymore.

File: src/Orleans.Serialization.Abstractions/Annotations.cs (hunk 1 above)
Comment (jsteinich) [reply src/Orleans.Serialization.Abstractions/Annotations.cs:49]: Certainly could and definitely would if there was a "real" dependency on it. As is, not intruding a dependency seems like the better option.

Side note: what are these invokers used for? I wasn't finding any references to them.

Comment (ReubenBond) [reply src/Orleans.Serialization.Abstractions/Annotations.cs:49]: Good point, it's unused now, so I've deleted it in the latest commit

Comment (ReubenBond) [reply src/Orleans.Serialization.Abstractions/Annotations.cs:49]: Could you select the option on this PR to allow me to push to your branch?

**Comment by ReubenBond** [comment]
I have a commit here implementing some suggestions: https://github.com/dotnet/orleans/commit/5b83257a3dacfaee9f743f574c29545ec113dded

* Rename the TargetFramework properties so that they have meaning again
* Remove the unused properties from `GenerateMethodSerializersAttribute`
* Fix the name of the `SystemMemoryData` property
* Remove the unnecessary Directory.Build.props

**Pull request merged**